cupy/core/raw.pyx

```diff
@@ -24,20 +22,21 @@ cdef class RawKernel:
         options (str): Compiler options passed to NVRTC. For details, see
             https://docs.nvidia.com/cuda/nvrtc/index.html#group__options.
         backend (str): Either `nvrtc` or `nvcc`. Defaults to `nvrtc`
+        translate_cucomplex (bool): Whether the CUDA source includes the header
+            `cuComplex.h` or not. If set to ``True``, any code that uses the
+            functions from `cuComplex.h` will be translated to its Thrust
+            counterpart. Defaults to ``False``.
     """
 
-    def __init__(self, code, name, options=(), backend='nvrtc'):
-        if isinstance(code, six.binary_type):
```

Comment:
If code is `bytes` type, it should be decoded even in Python 3. So these code cannot be removed.
(You can rewrite `six.binary_type` with `bytes`, but preferably do this in a separate PR as this is a different topic.)

Reply:
Didn't realize this 😢 I thought `code` is always a `str`?

Reply:
Generally we can assume that, but there's no reason to aggressively break backward compatibility. For example users may using binary stream to read file, e.g. `code=open('file.cu', 'rb').read()`.

Reply:
OK. But that's a really atypical/non-pythonic usage for reading a text-based file...